precompiles/erc20/integration_test.go

```diff
@@ -0,0 +1,1151 @@
+			DescribeTable("querying the name should return an error", func(callType int) {
+				txArgs, nameArgs := s.getTxAndCallArgs(callType, contractAddr, erc20.NameMethod)
+
+				noIBCVoucherCheck := execRevertedCheck
+				if callType == directCall {
+					noIBCVoucherCheck = failCheck.WithErrContains(
+						fmt.Sprintf("denom is not an IBC voucher: %s", s.tokenDenom),
+					)
+				}
+
+				_, _, err = s.factory.CallContractAndCheckLogs(sender.Priv, txArgs, nameArgs, noIBCVoucherCheck)
+				Expect(err).ToNot(HaveOccurred(), "unexpected result calling contract")
+			},
+				Entry(" - direct call", directCall),
+				Entry(" - through contract", contractCall),
+			)
+
+			DescribeTable("querying the symbol should return an error", func(callType int) {
+				txArgs, symbolArgs := s.getTxAndCallArgs(callType, contractAddr, erc20.SymbolMethod)
+
+				noIBCVoucherCheck := execRevertedCheck
+				if callType == directCall {
+					noIBCVoucherCheck = failCheck.WithErrContains(
+						fmt.Sprintf("denom is not an IBC voucher: %s", s.tokenDenom),
+					)
+				}
+
+				_, _, err = s.factory.CallContractAndCheckLogs(sender.Priv, txArgs, symbolArgs, noIBCVoucherCheck)
+				Expect(err).ToNot(HaveOccurred(), "unexpected result calling contract")
+			},
+				Entry(" - direct call", directCall),
+				Entry(" - through contract", contractCall),
+			)
+
+			DescribeTable("querying the decimals should return an error", func(callType int) {
+				txArgs, decimalsArgs := s.getTxAndCallArgs(callType, contractAddr, erc20.DecimalsMethod)
+
+				noIBCVoucherCheck := execRevertedCheck
+				if callType == directCall {
+					noIBCVoucherCheck = failCheck.WithErrContains(
+						fmt.Sprintf("denom is not an IBC voucher: %s", s.tokenDenom),
+					)
+				}
+
+				_, _, err = s.factory.CallContractAndCheckLogs(sender.Priv, txArgs, decimalsArgs, noIBCVoucherCheck)
+				Expect(err).ToNot(HaveOccurred(), "unexpected result calling contract")
+			},
+				Entry(" - direct call", directCall),
+				Entry(" - through contract", contractCall),
+			)
+		})
+
+		Context("for an IBC coin without registered metadata", func() {
+			const (
+				denom       = "uxmpl"
+				expName     = "Xmpl"
+				expSymbol   = "XMPL"
+				expDecimals = uint8(6)
+			)
+
+			BeforeEach(func() {
+				// Deploy new precompile for this test using the xmpl ibc voucher
+				//
+				// NOTE: this is not the same as the one used in the other tests
+				// because we need to set the metadata for this one.
+				ibcDenomTrace := types.DenomTrace{Path: "channel-0", BaseDenom: denom}
+
+				// overwrite the other precompile with this one, so that the test utils like s.getTxAndCallArgs still work.
+				s.precompile = s.setupERC20Precompile(ibcDenomTrace.IBCDenom())
+
+				// TODO: can I handle this differently, just using the integration utils and not using the keeper directly?
+				s.network.App.TransferKeeper.SetDenomTrace(s.network.GetContext(), ibcDenomTrace)
```

Comment:
this isn't ideal since we typically want to avoid using the actual keepers in the new integration test setup, but is there a way to add the denom trace to the transfer keeper in a transaction without having a separate IBC chain to send from? @facs95